lib/go/test/fuzz/fuzz.go

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// +build gofuzz
+
+package fuzz
+
+import (
+	"context"
+	"fmt"
+	"shared"
+	"strconv"
+	"tutorial"
+
+	"github.com/apache/thrift/lib/go/thrift"
+)
+
+const nbFuzzedProtocols = 2
+
+func fuzzChooseProtocol(d byte, t thrift.TTransport) thrift.TProtocol {
+	switch d % nbFuzzedProtocols {
+	case 0:
+		return thrift.NewTCompactProtocolFactoryConf(nil).GetProtocol(t)
+	}
+	return thrift.NewTBinaryProtocolFactoryConf(nil).GetProtocol(t)
```

Comment:
nit: The current code works, but it's hard to reason about. I'd suggest something like this:

```go
switch d % nbFuzzedProtocols {
default:
  fallthrough
case 0:
  return thrift.NewTBinaryProtocolFactoryConf(nil).GetProtocol(t)
case 1:
  return thrift.NewTCompactProtocolFactoryConf(nil).GetProtocol(t)
}
```

(note that I also swapped binary protocol and compact protocol, because binary really should be the default one).

Reply:
ok